src/main/java/org/commonjava/util/partyline/lock/local/ReentrantOperation.java

```diff
@@ -24,5 +29,6 @@
 public interface ReentrantOperation<T>
 {
     T execute( ReentrantOperationLock opLock )
-            throws InterruptedException, IOException;
+                    throws InterruptedException, IOException, SystemException, NotSupportedException,
```

Comment:
I would create a PartylineException class and throw that, then catch all of these and wrap them in PartylineException, with some contextual information about what was happening when the error occurred...what file was in play, what we were doing with it, etc.

We can even make PartylineException extend IOException so we can reuse the same throws clauses in calling code.

I think in the end we should throw PartylineException, IOException, InterruptedException from this.